sql/expression/function/strcmp_test.go

```diff
@@ -36,10 +37,13 @@ func TestStrCmp(t *testing.T) {
 		{"equal strings", sql.Text, sql.Text, sql.NewRow("a", "a"), int(0), nil},
 		{"first string is smaller", sql.Text, sql.Text, sql.NewRow("a", "b"), int(-1), nil},
 		{"second string is smaller", sql.Text, sql.Text, sql.NewRow("b", "a"), int(1), nil},
-		{"arguments have different types", sql.Int8, sql.Text, sql.NewRow(1, "1"), int(0), nil},
 		{"first argument is null", sql.Text, sql.Text, sql.NewRow(nil, "a"), nil, nil},
 		{"second argument is null", sql.Text, sql.Text, sql.NewRow("a", nil), nil, nil},
 		{"both arguments are null", sql.Text, sql.Text, sql.NewRow(nil, nil), nil, nil},
+		{"different collations throws error", sql.MustCreateString(sqltypes.Text, 10, 255), sql.MustCreateString(sqltypes.Text, 10, 309), sql.NewRow("a", "a"), nil, sql.ErrCollationIllegalMix},
```

Comment:
I'd prefer `sqltypes.VarChar` over `sqltypes.Text` here, since you're specifying a length of 10. `TEXT` types are defined on a boundary, so a `TEXT` type with length 10 and another with length 20 will resolve to a `TINYTEXT`.

In addition, **ALWAYS** prefer using `sql.Collation_xyz` instead of just passing in a number. It's not clear that the third parameter in `MustCreateString` is a collation, and this also does not lend itself nearly as well to refactoring. Not only that, some numbers are not used, so they're technically invalid collations.